typescript/core/src/McapWriter.ts

```diff
@@ -120,7 +129,60 @@ export class McapWriter {
     this.#compressChunk = compressChunk;
   }
 
+  /**
+   * Initializes a new McapWriter in append mode.
+   */
+  static async InitializeInAppendMode({
+    writable,
+    readable,
+  }: {
+    writable: IAppendWritable;
+    readable: IReadable;
```

Comment:
I'm not sure if there is specific prior art for this sort of API, but you could consider accepting these as a single object (if you are okay with forcing the caller to implement both interfaces on the same object).

You could also update the recently-added nodejs `FileHandleReadable`/`FileHandleWritable` class(es) so they can be used with this API.

Reply:
I'm not quite sure what you mean by this. Do you have an example?

Reply:
I meant like `IReadable & IAppendWritable` – since I'm not sure if I can imagine situations where the caller would actually want the reader and writer to be different objects?